server/src/main/scala/com/cloudera/livy/server/interactive/InteractiveSessionServlet.scala

```diff
@@ -195,8 +207,16 @@ class InteractiveSessionServlet(livyConf: LivyConf)
 
   jpost[AddResource]("/:id/add-jar") { req =>
     withSession { lsession =>
-      val uri = new URI(req.uri)
-      lsession.addJar(uri)
+      addJarOrPyFile(req, lsession)
+    }
+  }
+
+  jpost[AddResource]("/:id/add-pyfile") { req =>
+    withSession { lsession =>
+      lsession.kind match {
+        case Spark() => BadRequest("Only supported for pyspark sessions.")
```

Comment:
This should be a wildcard match after the pyspark match; you should have seen a warning here because your match is not exhaustive.